### ISSUE TITLE
Setup https using letsencrypt

### ISSUE THREAD

**Issue by captbaritone** [vim-awesome/vim-awesome#161]

**Comment by jonafato** [comment]
TL;DR: run the following commands for a quick and easy setup. Full details below the fold.

```
$ sudo apt-get update
$ sudo apt-get install software-properties-common
$ sudo add-apt-repository ppa:certbot/certbot
$ sudo apt-get update
$ sudo apt-get install python-certbot-nginx 
$ certbot --nginx --non-interactive --agree-tos --post-hook "service nginx reload" \
  --domains vimawesome.com --email webmaster@vimawesome.com  # Replace the email with a real one
$ certbot --renew --dry-run  # Test that renewal will work when the certbot cronjob (automatically installed) runs
```

For best results, modify the nginx config as described below to redirect all http requests to their https versions. As is, this gets a B on the [SSL Labs test](https://www.ssllabs.com/ssltest/); I'm looking into the config changes necessary to get this up to at least an A, but a B isn't a bad place to start.

----

The site is currently running behind nginx 1.4.6 on Ubuntu. Based on the [package page](https://packages.ubuntu.com/trusty/nginx), I'm guessing that it's Ubuntu 14.04. Using this, I grabbed the instructions from the [certbot documentation](https://certbot.eff.org/#ubuntutrusty-nginx). I set up a simple test environment at https://certbot-test.jonafato.com/ to match this stack. My initial nginx config looked like:

```nginx
server {
        listen 80 default_server;
        listen [::]:80 default_server ipv6only=on;

        server_name certbot-test.jonafato.com;

        root /usr/share/nginx/html;
        index index.html index.htm;

        location / {
                try_files $uri $uri/ =404;
        }
}
```

After installing and running certbot (and allowing it to update my config), it had new TLS settings:

```nginx
server {
        listen 80 default_server;
        listen [::]:80 default_server ipv6only=on;

        server_name certbot-test.jonafato.com;

        root /usr/share/nginx/html;
        index index.html index.htm;

        location / {
                try_files $uri $uri/ =404;
        }

        listen 443 ssl; # managed by Certbot
        ssl_certificate /etc/letsencrypt/live/certbot-test.jonafato.com/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/certbot-test.jonafato.com/privkey.pem; # managed by Certbot
        include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
}
```

To make sure that TLS is always used, modify that server block:

1. Split it into two server blocks, one for http and one for https.
2. Configure the http server block to redirect to the https version of the site.

This gives us a version of the config that support and enforces https:

```nginx
server {
        listen 80 default_server;
        listen [::]:80 default_server ipv6only=on;

        server_name certbot-test.jonafato.com;

        # Redirect all non-https traffic
        location / {
            return 301 https://$host$request_uri;
        }

}

server {
        listen 443 ssl; # managed by Certbot
        server_name certbot-test.jonafato.com;

        root /usr/share/nginx/html;
        index index.html index.htm;

        location / {
                try_files $uri $uri/ =404;
        }

        ssl_certificate /etc/letsencrypt/live/certbot-test.jonafato.com/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/certbot-test.jonafato.com/privkey.pem; # managed by Certbot
        include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
}
```

**Comment by captbaritone** [comment]
Nice! This worked perfectly! We should be exclusively https now. Sorry it took me two weeks to get to this.
